src/main/java/com/softserve/demo/service/impl/OrderServiceImpl.java

```diff
@@ -38,8 +43,16 @@ public Order createOrder(Order order) {
         return orderRepository.findAll();
     }
 
+    @Override
+    public Order updateOrder(Order order) {
+        orderRepository.findById(order.getId()).orElseThrow(() ->
+                new NotFoundException(String.format("Order with id: [%d] not found", order.getId())));
+        return orderRepository.save(order);
+    }
+
     @Override
     public void deleteOrder(Integer id) {
+        log.debug(String.format("Delete order with id: [%d]", id));
```

Comment:
Here could be the case when the order which you want to delete doesn't exist. But you will log that it was deleted. But it wasn't.